src/serializer/Referentializer.js

```diff
@@ -161,8 +164,10 @@ export class Referentializer {
   _getReferentializationScope(residualBinding: ResidualFunctionBinding): ReferentializationScope {
     if (residualBinding.potentialReferentializationScopes.has("GLOBAL")) return "GLOBAL";
     if (residualBinding.potentialReferentializationScopes.size > 1) {
-      // TODO #2428: Revisit for nested optimized functions.
```

Comment:
This is where the actual bugfix is. The rest of the PR is refactoring to make `tryGetOutermostOptimizedFunction` availabe to this file.